coffee-chats/src/main/webapp/src/util/chatRequest.ts

```diff
@@ -25,19 +26,22 @@ export const submitCalAuthRequest = async () => {
  * indicate whether the chat request has been fulfilled yet).
  * 
  * @param interests - Array of user inputted tags they prefer to speak about.
- * @param dates - Array of dates user wants to speak on.
+ * @param startDates - Array of starting datetimes user wants to speak on.
+ * @param endDates - Array of ending datetimes user wants to speak on.
  * @param numPeopleRange - Array of two numbers with the range of number of participants.
  * @param durationMins - Duration of the requested chat in minutes.
  * @param matchRandom - Boolean to still match if no interest matches are found.
  * @param matchRecents - Boolean to still match with people spoken to recently.
  */
-export const submitChatRequest = async (interests: string[], dates: Date[],
-    numPeopleRange: number[], durationMins: number, matchRandom: boolean, 
+export const submitChatRequest = async (interests: string[],
+    startDates: MaterialUiPickersDate[], endDates: MaterialUiPickersDate[],
+    numPeopleRange: number[], durationMins: number, matchRandom: boolean,
     matchRecents: boolean) => {
   const data = new URLSearchParams();
   
   data.append('tags', interests.toString());
-  data.append('dates', dates.map(d => +d).toString());
+  data.append('startDates', startDates.map(d => d!).map(d => +d).toString());
```

Comment:
what does d! do?

Reply:
The material UI component has a wrapper type for date so that different types can be used, but it is defined as Date or null. `d!` is like performing a get on an optional in that it returns the object as the non-null/undefined type. I have already checked them to be not null before function call so this shouldn't raise any errors.

Reply:
Why not `+d!`?

Reply:
That should work I think, I am not sure about the binding strength of the operators, but yes it should work in one step